### DISCUSSION TITLE
Lazy loading and displaying additional data

### DISCUSSION THREAD

**Discussion by kpalmvik** [remix-run/remix#1314] [Q&A] [answered]
Hi!

I would like to lazy load and display additional data on the client, and not include this data in the server generated HTML.

For example, imagine a blog post followed by a “related posts” section. The blog post text would obviously be server rendered, but finding the related posts might be magnitudes slower and should therefore not block the initial page load.

Would the recommended approach be to create a resource route and fetch the data manually in a component (through `useEffect` or similar), or is there an obvious Remix solution that I have missed?

**Comment by sergiodxa** [comment, marked as answer]
You should create the resource route as you said and use the useFetcher hook to fetch it

**Reply by kpalmvik** [reply]
Thank you for the answer, @sergiodxa!

Relevant documentation for reference:
- [Resource routes](https://remix.run/docs/en/v1/guides/resource-routes)
- [`useFetcher`](https://remix.run/docs/en/v1/api/remix#usefetcher)
